### PR TITLE
#163512197: Users can create,update and delete comments

### PR DESCRIPTION
**Pull request by davishooly**
**What does this PR do?**
-  Implements user ability to delete comments on articles.
- Implements user ability to update comments on articles.

**Description of tasks to be completed?**
- Authenticated users can delete and update the comments they have posted.

**How should this be manually tested?**

- Clone the repo and cd into the project folder
- Switch to the appropriate branch with `feedback-comments-163512197`
- Run `npm install` to update your packages.
- Run tests with `npm test`
- Specify the domain you are running the app from with `export REACT_APP_API="https://ah-codetitans-staging.herokuapp.com/"`
- Fire up the server with `npm start`
- Login to the app.
- Navigate to `/article/best-python-version` and post a new comment on the article.
- Once the single article loads, click on the share buttons to share.

**What are the relevant pivotal tracker stories?**
- [#163512197](https://www.pivotaltracker.com/story/show/163512197)

**Checklist (edited)**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

**Screenshots**
screen shot 2019-02-06 at 19 25 02

### CODE REVIEW

**Review by denomash — APPROVED** [review]
👍

**Review by Darius-Ndubi — APPROVED** [review]
👍

**Review by Bryan-Cee — APPROVED** [review]
👍 LGTM

File: src/setupTests.js [hunk 1]
```diff
@@ -3,20 +3,17 @@ import React from "react";
 import { configure, mount } from "enzyme";
 import Adapter from "enzyme-adapter-react-16";
 import { Provider } from "react-redux";
-import { ConnectedRouter } from "connected-react-router";
-
-import configureStoreDev, { history } from "./store/configureStore";
+import { MemoryRouter } from "react-router";
+import configureStore, { configureStoreTest } from "./store/configureStore";
```
Comment (justmesam) [inline src/setupTests.js]: why is `configureStoreTest` imported here??

Comment (davishooly) [reply src/setupTests.js]: I have removed the unused import

**Review by bettblake08 — APPROVED** [review]
👍

**Review by justmesam — APPROVED** [review]

**Pull request merged**